src/pages/SupplierInvoicePage.js

```diff
@@ -1,398 +1,328 @@
-   * @param   {object}  transactionBatch  The transaction batch from the row being edited.
-   * @param   {string}  newValue          The value the user entered in the cell.
-   * @return  {none}
-   */
-  onEndEditing = (key, transactionBatch, newValue) => {
-    const { database } = this.props;
-    database.write(() => {
-      switch (key) {
-        case 'totalQuantity':
-          // Should edit |numberOfPacks| directly if |packSize| becomes an editable column
-          // that represents the number of packs counted.
-          transactionBatch.setTotalQuantity(database, parsePositiveInteger(newValue));
-          break;
-        case 'expiryDate':
-          transactionBatch.expiryDate = newValue;
-          break;
-        default:
-          break;
-      }
-      database.save('TransactionBatch', transactionBatch);
-    });
-  };
-
-  getModalTitle = () => {
-    const { modalKey } = this.state;
-
-    const { ITEM_SELECT, COMMENT_EDIT, THEIR_REF_EDIT } = MODAL_KEYS;
-    switch (modalKey) {
-      default:
-      case ITEM_SELECT:
-        return modalStrings.search_for_an_item_to_add;
-      case COMMENT_EDIT:
-        return modalStrings.edit_the_invoice_comment;
-      case THEIR_REF_EDIT:
-        return modalStrings.edit_their_reference;
-    }
-  };
-
-  updateDataFilters = (newSearchTerm, newSortBy, newIsAscending) => {
-    // (... != null) checks for null or undefined (implicitly type coerced to null).
-    if (newSearchTerm != null) this.dataFilters.searchTerm = newSearchTerm;
-    if (newSortBy != null) this.dataFilters.sortBy = newSortBy;
-    if (newIsAscending != null) this.dataFilters.isAscending = newIsAscending;
-  };
-
-  /**
-   * Returns updated data fitlered by |searchTerm| and ordered by |sortBy| and |isAscending|.
-   */
-  refreshData = (newSearchTerm, newSortBy, newIsAscending) => {
-    this.updateDataFilters(newSearchTerm, newSortBy, newIsAscending);
-    const { searchTerm, sortBy, isAscending } = this.dataFilters;
-    const { database, transaction } = this.props;
-    const transactionBatches = transaction
-      .getTransactionBatches(database)
-      .filtered('itemName BEGINSWITH[c] $0', searchTerm);
-
-    const sortDataType = SORT_DATA_TYPES[sortBy] || 'realm';
-
-    this.setState({
-      data: sortDataBy(transactionBatches, sortBy, sortDataType, isAscending),
-    });
-  };
-
-  addNewLine = item => {
-    const { database, transaction } = this.props;
-    database.write(() => {
-      const transactionItem = createRecord(database, 'TransactionItem', transaction, item);
-      createRecord(
-        database,
-        'TransactionBatch',
-        transactionItem,
-        createRecord(database, 'ItemBatch', item, '')
+  }, []);
+
+  const renderPageInfo = useCallback(
+    () => <PageInfo columns={pageInfo(pageObject, dispatch)} isEditingDisabled={isFinalised} />,
+    [comment, theirRef, isFinalised]
+  );
+
+  const renderCells = useCallback(
```

Comment:
All of this has been factored out in #1163 - sorry :)

Reply:
all g. I approved #1163